src/main/groovy/edu/oregonstate/mist/trailsapi/db/TrailDAO.groovy

```diff
@@ -134,5 +134,37 @@ interface TrailDAO extends Closeable {
 		 LEFT JOIN TRAIL_DIFFICULTIES ON TRAILS.DIFFICULTY_ID = TRAIL_DIFFICULTIES.DIFFICULTY_ID
 		 WHERE TRAILS.ID = :id
 	""")
-	Trail getTrailByID(@Bind("id") Integer id)
+    Trail getTrailByID(@Bind("id") Integer id)
+
+    /***********************************************************************************************
+    PUT /trails/{trailID}
+    Updates values if binded value is not null
+    ***********************************************************************************************/
+    @SqlUpdate("""
+        UPDATE TRAILS
+        SET
+            NAME = COALESCE(:name, NAME),
+            DIFFICULTY_ID = COALESCE((SELECT DIFFICULTY_ID FROM TRAIL_DIFFICULTIES
+            WHERE DIFFICULTY_COLOR = :difficulty), DIFFICULTY_ID),
+            ZIPCODE = COALESCE(:zipCode, ZIPCODE),
+            SMALLDROP = COALESCE(:smallDrop, SMALLDROP),
+            LARGEDROP = COALESCE(:largeDrop, LARGEDROP),
+            WOODRIDE = COALESCE(:woodRide, WOODRIDE),
+            SKINNY = COALESCE(:skinny, SKINNY),
+            LARGEJUMP = COALESCE(:largeJump, LARGEJUMP),
```

Comment:
By using these coalesce functions, you are sort of doing more of an implementation of `PATCH` than `PUT` - which is fine, but if so, you might want to reflect that in your resource and swagger spec. For `PUT` these coalesce functions would conflict because I don't think it would allow me to update a field to null. However, if you were doing `PATCH`, you would still want the ability to make a field null (if that's allowed in the first place, given your API design).